src/cmd/troubleshoot/checks_executor.go

```diff
@@ -0,0 +1,48 @@
+package troubleshoot
+
+import (
+	tserrors "github.com/Dynatrace/dynatrace-operator/src/cmd/troubleshoot/errors"
+)
+
+type Result int
+
+const (
+	PASSED = iota + 1
+	FAILED
+	SKIPPED
+)
+
+var results = map[string]Result{}
+
+func runChecks(troubleshootCtx *troubleshootContext, checks []Check) error {
+	errs := tserrors.NewAggregatedError()
+	for _, check := range checks {
+		if shouldSkip(check) {
+			results[check.Name] = SKIPPED
+			continue
+		}
+
+		if err := check.Do(troubleshootCtx); err != nil {
+			logErrorf(err.Error())
+			errs.Add(err)
+			results[check.Name] = FAILED
+		} else {
+			results[check.Name] = PASSED
+		}
+	}
+
+	if errs.Empty() {
+		return nil
+	}
+
+	return errs
+}
+
+func shouldSkip(check Check) bool {
```

Comment:
Reverse invers logic and improve naming
```suggestion
func prerequisitesHavePassed(check Check) bool {
```

Reply:
IMO `shouldSkip` is fine as there can be more reason to skip (in the future maybe?) 
also `prerequisitesPassed` is enough

Reply:
In the calling function `shouldSkip` is better understandable.
`if shouldSkip(check)` is more straight forward compared to `if !prerequisitesPassed(check)`